geom/src/polygon.rs

```diff
@@ -476,19 +476,38 @@ fn to_geo(pts: &Vec<Pt2D>) -> geo::Polygon<f64> {
     )
 }
 
-fn from_geo(p: geo::Polygon<f64>) -> Polygon {
-    Polygon::buggy_new(
-        p.into_inner()
-            .0
-            .into_points()
+impl From<geo::Polygon<f64>> for Polygon {
+    fn from(poly: geo::Polygon<f64>) -> Self {
+        Polygon::buggy_new(
+            poly.into_inner()
+                .0
+                .into_points()
+                .into_iter()
+                .map(|pt| Pt2D::new(pt.x(), pt.y()))
+                .collect(),
+        )
+    }
+}
+
+impl From<Polygon> for geo::Polygon<f64> {
+    fn from(poly: Polygon) -> Self {
+        let exterior_coords = poly
+            .points
             .into_iter()
-            .map(|pt| Pt2D::new(pt.x(), pt.y()))
-            .collect(),
-    )
+            .map(geo::Coordinate::from)
+            .collect::<Vec<_>>();
+        let exterior = geo::LineString(exterior_coords);
+
+        let interiors: Vec<geo::LineString<f64>> = poly
+            .rings
+            .map(|rings| rings.into_iter().map(geo::LineString::from).collect())
```

Comment:
`Polygon` only has half support for holes... preserving the original rings (both exterior and interior) is nice for generating outlines later. Places calling `buggy_new` or `precomputed` (notably `PolyLine::make_polygons`) don't fill out the rings.

If rings is set, then [0] is the exterior, correct. `points` contain every point in both the exterior and interior, with `indices` defining the triangles.

I wonder if it would be simpler and maybe more performant to just make `Polygon` store an exterior Ring and some interior Rings, and delay triangulation entirely until it's needed for rendering. `contains_pt` also uses the triangles now, but there might be other algorithms for figuring that out without triangulating. Maybe `geom::Polygon` could actually just be a zero-cost wrapper around `geo::Polygon`! Disclaimer, I haven't thought through these ideas.

Reply:
Thanks for your thoughts. 

I think it'd be nice to be uniform about where the exterior lives. either always polygon.rings[0] or as a new polygon.exterior.

> Maybe geom::Polygon could actually just be a zero-cost wrapper around geo::Polygon
I am in theory on board with this!
> I wonder if it would be simpler and maybe more performant to just make Polygon store an exterior Ring and some interior Rings, and delay triangulation entirely until it's needed for rendering

I haven't spent time surveying the existing usage of geom::polygon - but maybe it makes sense to have separate entities for "polygon" vs. "polygon that I'm going to draw"

I'm not intending to change anything drastically in this PR, but I'll try to keep it in mind as I work with more of the existing geometry code in ABStreet.